crates/sui-rpc-api/proto/sui.types.proto

```diff
@@ -0,0 +1,1040 @@
+syntax = "proto3";
+
+package sui.types;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+message Address {
+  bytes address = 1;
+}
+
+message ObjectId {
+  bytes object_id = 1;
+}
+
+message Digest {
+  bytes digest = 1;
+}
+
+message Bcs {
+  bytes bcs = 1;
+  // optional bytes name = 2;
+}
+
+// Checkpoint {{{
+
+message CheckpointSummary {
+  // Reserve field 1 and 2 for possible bcs bytes and/or Digest of type
+  reserved 1, 2;
+
+  uint64 epoch = 3;
+  uint64 sequence_number = 4;
+  uint64 total_network_transactions = 5;
+  Digest content_digest = 6;
+  optional Digest previous_digest = 7;
+
+  GasCostSummary epoch_rolling_gas_cost_summary = 8;
+
+  google.protobuf.Timestamp timestamp = 9;
+
+  repeated CheckpointCommitment commitments = 10;
+  optional EndOfEpochData end_of_epoch_data = 11;
+  bytes version_specific_data = 12;
+}
+
+message CheckpointCommitment {
+  oneof commitment {
+    Digest ecmh_live_object_set = 1;
+  }
+}
+
+message EndOfEpochData {
+  repeated ValidatorCommitteeMember next_epoch_committee = 1;
+  uint64 next_epoch_protocol_version = 2;
+  repeated CheckpointCommitment epoch_commitments = 3;
+}
+
+message CheckpointedTransactionInfo {
+  // TransactionDigest
+  Digest transaction = 1;
+  // EffectsDigest
+  Digest effects = 2;
+  repeated UserSignature signatures = 3;
+}
+
+message CheckpointContents {
+  message V1 {
+    repeated CheckpointedTransactionInfo transactions = 1;
+  }
+
+  // Reserve field 1 and 2 for possible bcs bytes and/or Digest of type
+  reserved 1, 2;
+
+  oneof contents {
+    V1 v1 = 3;
+  }
+}
+
+// End Checkpoint }}}
+
+// TransactionEvents {{{
+
+message TransactionEvents {
+  // Reserve field 1 and 2 for possible bcs bytes and/or Digest of type
+  reserved 1, 2;
+
+  repeated Event events = 3;
+}
+
+message Event {
+  ObjectId package_id = 1;
+  Identifier module = 2;
+  Address sender = 3;
+  StructTag event_type = 4;
+  bytes contents = 5;
+  // rendered
```

Comment:
what does "rendered" mean?
 (in a few places)

Reply:
ah, this was a placeholder comment about areas where we may want to add a field later, for example rendering an event into a parsed more human readable format vs just the BCS.